### ISSUE TITLE
My flatpak installation lost the posibillity to restrict filesystem access after a 'flatpak update' run

### ISSUE THREAD

**Issue by schwidom** [flatpak/flatpak#4399]
## Linux distribution and version
Debian 8 / Debian 10 with all possible packages updated to the Debian 10 version

uname -a :
Linux debian64 4.9.0-0.bpo.9-amd64 #1 SMP Debian 4.9.168-1+deb9u5~deb8u1 (2019-08-13) x86_64 GNU/Linux

lsb_release -a :
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 10 (buster)
Release:	10
Codename:	buster

## Flatpak version
flatpak --version :
Flatpak 1.2.5

LANG=C dpkg -l '*flatpak*'| grep '^ii' :
ii  flatpak                       1.2.5-0+deb10u4 amd64

## Description of the problem / ## Steps to reproduce
Before the problem I was able to restrict the filesystem usage of org.mozilla.firefox (and other apps) via com.github.tchx84.Flatseal.
Then I installed also org.signal.Signal and org.telegram.desktop and I checked always the filesystem restriction. It worked.
After running 'flatpak update' : firefox  and the other apps are still running but the filesystem restriction does not longer work.
The configuration in github.tchx84.Flatseal is how I left it and I can make filesystem access changes in this tool and it gets written in ~/.local/share/flatpak/overrides/<appname> as usual but it has no longer effect.
I can restrict network access and pulse audio and x11 / wayland usage but not the filesystem usage.
With firefox I can access all other paths read and write, before that It wasn't possible.

flatpak list --app --columns=application,version
...
com.github.tchx84.Flatseal                             1.7.4
org.signal.Signal                                      5.16.0
org.telegram.desktop                                   3.0.1
org.mozilla.firefox                                    91.0.2

$ cat .local/share/flatpak/overrides/org.mozilla.firefox
[Context]
devices=!all
filesystems=!/etc/firefox/policies;!xdg-download;!xdg-run/pipewire-0

[System Bus Policy]
org.freedesktop.NetworkManager=none

[Session Bus Policy]
org.freedesktop.Notifications=none
org.freedesktop.FileManager1=none
org.a11y.Bus=none
org.freedesktop.ScreenSaver=none
org.gtk.vfs.*=none
org.gnome.SessionManager=none
org.mozilla.firefox.*=none
org.mpris.MediaPlayer2.firefox.*=none

In the meantime I was able to compile version 1.9.3 of flatseal by myself but it does not work in a direct way with debian 10 by 'make install'. 

I would be glad if there were a script, which utilizes ./configure in a way so that it matches to the debian 10 needs. Maybe this is my solution.

Thanks in Advance,
Frank Schwidom

**Comment by refi64** [comment]
Flatpak 1.2 is *very* old at this point, if you use the version in [backports](https://backports.debian.org/Instructions/) does it work?

**Comment by schwidom** [comment]
It is the same as before. I installed almost all flatpak related  packets from backports :

$ dpkg -l '*flatpak*' | grep '^ii'
ii  flatpak                       1.10.2-1~bpo10+1 amd64        Application deployment framework for desktop apps
ii  flatpak-tests                 1.10.2-1~bpo10+1 amd64        Application deployment framework for desktop apps (tests)
ii  gir1.2-flatpak-1.0:amd64      1.10.2-1~bpo10+1 amd64        Application deployment framework for desktop apps (introspection)
ii  libflatpak-dev:amd64          1.10.2-1~bpo10+1 amd64        Application deployment framework for desktop apps (development)
ii  libflatpak-doc                1.10.2-1~bpo10+1 all          Application deployment framework for desktop apps (documentation)
ii  libflatpak0:amd64             1.10.2-1~bpo10+1 amd64        Application deployment framework for desktop apps (library)

$ dpkg -l '*bubblewrap*' | grep '^ii'
ii  bubblewrap     0.4.1-3~bpo10+1 amd64        setuid wrapper for unprivileged chroot and namespace manipulation


and it behaves exactly the same.

Regards.

**Comment by schwidom** [comment]
The problem seems to be no problem because I tested the filesysaccess via the file dialog.
Obviously the filedialig creates the right to load / save content at specific locations. 
By checking the access rights via file:///... it became clear that not all is allowed.
But I registered definitely a change in behaviour by updating all flatpak app and runtime packages.
The filedialog before that did not show all files in home.
Anyway. It is an unconventional way of restricting access, compared with lets say tomoyo or firejail. But I like the concept. It would be cool if it were better documented at the point when it happens.
It would be also cool if this approach could also work for the URL. Only the URL which I input were allowed then.

Thanks for reading,
Frank Schwidom
